### PR TITLE
Feat/audit2

### PR DESCRIPTION
**Pull request by sh-cha**
## Release Notes

- **New Features**
	- Introduced a new function to check for missing attributes in multiple parsing functions, enhancing error reporting.
	- Added a new field to track transaction success in the transaction handler.
	- Implemented a sleep mechanism with exponential backoff for retry logic.

- **Improvements**
	- Enhanced transaction handling by ignoring failed transactions.
	- Simplified withdrawal initiation process, improving maintainability.
	- Enhanced error handling with a backoff strategy for transaction processing.

- **Bug Fixes**
	- Updated transaction timeout checks to ensure timely processing of pending transactions.

- **Chores**
	- Adjusted file permission settings for batch file access.

### CODE REVIEW

File: provider/child/parse.go [hunk 1]
```diff
@@ -11,12 +11,32 @@ import (
...
 	from, to, baseDenom string,
 	amount sdk.Coin,
 	err error) {
+	missingAttrs := map[string]struct{}{
+		opchildtypes.AttributeKeyL1Sequence: {},
+		opchildtypes.AttributeKeySender:     {},
+		opchildtypes.AttributeKeyRecipient:  {},
+		opchildtypes.AttributeKeyDenom:      {},
+		opchildtypes.AttributeKeyBaseDenom:  {},
+		opchildtypes.AttributeKeyAmount:     {},
```
Comment (this-is-chainlight) [inline provider/child/parse.go]: `AttributeKeyFinalizeHeight` is missing I guess.

File: provider/child/parse.go [hunk 2]
```diff
@@ -11,12 +11,33 @@ import (
 	sdk "github.com/cosmos/cosmos-sdk/types"
 )
 
+func missingAttrsError(missingAttrs map[string]struct{}) error {
+	if len(missingAttrs) != 0 {
+		missingAttrStr := ""
+		for attr := range missingAttrs {
+			missingAttrStr += attr + " "
+		}
+		return fmt.Errorf("missing attributes: %s", missingAttrs)
```
Comment (beer-1) [inline provider/child/parse.go]: seems need to print missingAttrStr

File: provider/host/parse.go [hunk 3]
```diff
@@ -2,21 +2,41 @@ package host
...
 	ophosttypes "github.com/initia-labs/OPinit/x/ophost/types"
 )
 
+func missingAttrsError(missingAttrs map[string]struct{}) error {
+	if len(missingAttrs) != 0 {
+		missingAttrStr := ""
+		for attr := range missingAttrs {
+			missingAttrStr += attr + " "
+		}
+		return fmt.Errorf("missing attributes: %s", missingAttrs)
```
Comment (beer-1) [inline provider/host/parse.go]: Need to print missingAttrStr

File: types/retry.go [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+import (
+	"context"
+	"math"
+	"math/rand/v2"
+	"time"
+)
+
+func SleepWithRetry(ctx context.Context, retry int) {
+	// to avoid to sleep too long
+	if retry > 7 {
```
Comment (beer-1) [inline types/retry.go]: good to make this 7 as constant variable and use it in other files.

File: types/retry.go [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+	} else if retry == 0 {
+		return
+	}
+
+	sleepTime := 2 * math.Exp2(float64(retry))
+	sleepTime += rand.Float64() * float64(sleepTime) * 0.5 //nolint:all
+	timer := time.NewTimer(time.Duration(sleepTime) * time.Second)
+	select {
+	case <-ctx.Done():
+		return
```
Comment (beer-1) [inline types/retry.go:26]: good to add timer.Stop() as coderabbitai said

**Review by beer-1 — APPROVED** [review]
LGTM

**Pull request merged**